### PR TITLE
Implement Array.prototype.pop() in JS

### PR DESCRIPTION
**Pull request by cpq**

### CODE REVIEW

File: src/stdlib.c [hunk 1]
```diff
@@ -231,26 +231,33 @@ static void init_js_stdlib(struct v7 *v7) {
...
-      return i;
-    }
-  }
-  return -1;
-};
-));
+    Array.prototype.indexOf = function(a, b) {
+      if (!b || b < 0) b = 0;
+      for (var i = b; i < this.length; i++) {
+        if (this[i] == a) {
```
Comment (mmikulicic) [inline src/stdlib.c]: noticed now: strict equality

File: src/stdlib.c [hunk 2]
```diff
@@ -231,26 +231,33 @@ static void init_js_stdlib(struct v7 *v7) {
...
+        }
+      }
+      return -1;
+    };));
+
+  v7_exec(v7, &res, STRINGIFY(
+    Array.prototype.lastIndexOf = function(a, b) {
+      if (!b || b < 0 || b >= this.length) b = this.length - 1;
+      for (var i = b; i >= 0; i--) {
+        if (this[i] == a) {
```
Comment (mmikulicic) [inline src/stdlib.c]: same here

File: src/stdlib.c [hunk 3]
```diff
@@ -231,26 +231,33 @@ static void init_js_stdlib(struct v7 *v7) {
...
+        }
+      }
+      return -1;
+    };
+    ));
+
+  v7_exec(v7, &res, STRINGIFY(
+    Array.prototype.pop = function(a, b) {
+      var i = this.length - 1;
+      return this.splice(i, 1)[0];
```
Comment (mmikulicic) [inline src/stdlib.c]: popping an empty array should return `undefined`, not an empty array.

Comment (cpq) [inline src/stdlib.c]: It is undefined, isn't it? [][0] returns undefined.

Comment (mmikulicic) [inline src/stdlib.c]: Oh yes, sorry

**Comment by cpq** [comment]
Addressed strict equality comments, PTAL

**Pull request merged**